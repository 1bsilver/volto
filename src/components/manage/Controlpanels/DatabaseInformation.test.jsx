import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import DatabaseInformation from './DatabaseInformation';

const mockStore = configureStore();

jest.mock('react-portal', () => ({
  Portal: jest.fn(() => <div id="Portal" />),
}));
jest.mock('../../theme/Comments/CommentEditModal', () =>
  jest.fn(() => <div id="modal" />),
);

describe('DatabaseInformation', () => {
  it('renders a database information component', () => {
    const store = mockStore({
      controlpanels: {
        controlpanel: [
          {
            '@id': 'http://localhost:8080/Plone/@controlpanels/security',
            group: 'Security',
            title: 'Security',
          },
        ],
      },
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <DatabaseInformation location={{ pathname: '/database' }} />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
