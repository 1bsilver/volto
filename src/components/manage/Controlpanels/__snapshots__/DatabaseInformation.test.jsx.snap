// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatabaseInformation renders a database information component 1`] = `null`;
